primitives/state-machine/src/ext.rs

```diff
@@ -375,12 +360,6 @@ where
 		key: StorageKey,
 		value: Option<StorageValue>,
 	) {
-		trace!(target: "state", "{:04x}: PutChild({}) {}={:?}",
```

Comment:
Why are these removed?

Reply:
They are not necessary for our goals with balance reconciliation and we wanted to reduce payload size.

Reply:
All these are required for debugging. You could for example only change the level of the two you are interested in to debug?

Or maybe add some special tag (or whatever these extra arguments to the tracing log macros are called), so you can filter them on your own?

Reply:
I brought back the old traces and changed the target for the traces we need from `state` => `state_get_put`, this way if someone wants all the trace events in this file they can use `state` as the target and since it matches on prefix it will gather events on both targets.

cc @mattrutherford

Reply:
Seems like the most simple soluition. The alternative would be to change the filtering to be more fine grained, by incorporating name and fields as well as target, as is used by tracing_subscriber EnvFilter:

https://docs.rs/tracing-subscriber/0.2.17/tracing_subscriber/filter/struct.EnvFilter.html

If we wanted to be compatible with that, then the syntax to match would I think be something like `state{Method=\"get\"}` etc...  which could be passed as the RPC params or set as a default

Reply:
I would prefer it if we made the filtering more fine grained as @mattrutherford suggests. Relying on the target string is a bit brittle.

Reply:
I tried out directive filtering in commit `6076975` with the expression `state[{method}]`; this correctly filtered the events but it also filtered out all the spans. Since we already have custom prefix based filtering on targets (i.e `metadata.target().starts_with(target`)I did not want to mess around with creating more directives based on the user supplied targets.

Instead I added a check on events our `Subscriber::enabled` impl.
Will comment there to point it out.
I am not convinced though that this is more stable since we still have rely on a particular field existing which seems as brittle as target; and in the current setup I have it gives the users less options.

cc @dvdplm @mattrutherford